ENIGMAsystem/SHELL/Graphics_Systems/OpenGL-Common/shader.cpp

```diff
@@ -96,9 +96,15 @@
     enigma::ShaderProgram* ptiter = enigma::shaderprograms[program];
 #endif
 
+#if defined(__ANDROID__) && !defined(GL_GEOMETRY_SHADER)
```

Comment:
From here down: If these constants aren't guaranteed by the GL spec, then fine; that means this has nothing to do with Android. Remove the Android check. If they *are* guaranteed by the GL spec, but not the Android spec, then define them in a proxy header and revert this section of the file. Most of the below are just well-defined integers. If we need an `ENIGMA_NO_GEOMETRY_SHADERS`, or whatever, then fine.

Reply:
I agree with Josh, our last attempt was a lot cleaner fundies. I too worry about people adding desktop features and breaking Android more than people adding desktop features and Android just not having them immediately. Feel free to divide functions as granular as you need to.